src/NHibernate/Criterion/Expression.cs

```diff
@@ -18,57 +18,72 @@ private Expression()
 
 		/// <summary>
 		/// Apply a constraint expressed in SQL, with the given SQL parameters
+		/// The string {alias} will be replaced by the alias of the root entity.
+		/// Criteria aliases can also be used: "{a}.Value + {bc}.Value". Such aliases need to be registered via call to AddCriteriaAliases("a", "bc")
 		/// </summary>
 		/// <param name="sql"></param>
 		/// <param name="values"></param>
 		/// <param name="types"></param>
 		/// <returns></returns>
-		public static AbstractCriterion Sql(SqlString sql, object[] values, IType[] types)
+		public static SQLCriterion Sql(SqlString sql, object[] values, IType[] types)
```

Comment:
I don't know... How about deprecating this class altogether and create `Restrictions.Sql` instead?

Considering class summary  `This class is semi-deprecated. Use <see cref="Restrictions"/>. ` might be a good idea..

Reply:
Ok.. Might not be so good idea of deprecating full `Expression` class. At least not in scope of this                           PR. It's used quite extensively across code instead of recommended `Restrictions`. So how about deprecate only `Expression.Sql` methods?

Reply:
Most usages are in tests, only 6 in NHibernate. It sound a bit strange to deprecate all its methods but not the class itself.
But well, if you prefer to deprecate the class in another PR, why not.

Reply:
It's actively used in documentation too. So I think it's better to keep this class be empty for legacy reasons..

Reply:
Done

Reply:
We can still obsolete it, but we will likely have to keep it in next major version. Not every obsoleted members are meant to be dropped, for easing upgrades, especially those not costly to keep and potentially massively used. (That is also the version for telling in code since which version the member is obsoleted, helping to decide whether to drop it or not. Otherwise if we drop them all on each major version, the comment would not be useful.)

From [Microsoft guidelines](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/breaking-changes#binary-breaking-change):

>CONSIDER keeping types and methods with the `ObsoleteAttribute` indefinitely in low and middle-level libraries.

(They classify an ORM as a low or middle-level library.)